docs/src/checklists.md

```diff
@@ -0,0 +1,38 @@
+# Checklists
+
+The purpose of this page is to collate a series of checklists for commonly
+performed changes to the source code of Documenter.
+
+In each case, copy the checklist into the description of the pull request.
+
+## Making a release
+
+In preparation for a release, use the following checklist.
+
+````
+## Pre-release
+
+ - [ ] Change the version number in `Project.toml`
+        * If the release is breaking, increment MAJOR
+        * If the release adds a new user-visible feature, increment MINOR
+        * Otherwise (bug-fixes, documentation improvements), increment PATCH
+ - [ ] Update `CHANGELOG.md`, following the existing style (in particular, make
+       sure that the change log for this version has the correct version number
+       and date).
+ - [ ] Run `make changelog`, to make sure that all the issue references in
+       `CHANGELOG.md` are up to date.
+ - [ ] Check that the commit messages in this PR do not contain `[ci skip]`
+
+## The release
+
+ - [ ] After merging this pull request, tag the release. There are two options for this:
```

Comment:
Again, what is "this pull request"? If it's the PR for the release branch, I would usually tag the release *before* merging the PR. This is in case there is any kind of problem like a broken compat bound that might only show up in the registration itself. If there is a problem, I would amend the release branch, force push it, and retrigger the registration.

Once the registration goes through and Tagbot tags the commit, I would then merge the PR (using a fast-forward merge)

The one annoyance I have with merging the release-PR *after* the release is tagged is that it then shows up as a "closed PR" for the *next* release in the release notes that Github auto-generates.